lib/tapioca/rbi/rewriters/sort_nodes.rb

```diff
@@ -11,11 +11,11 @@ class SortNodes < Visitor
         def visit(node)
           return unless node.is_a?(Tree)
           visit_all(node.nodes)
+          original_order = original_nodes_order(node.nodes.dup)
           node.nodes.sort! do |a, b|
-            return 0 if a.is_a?(Mixin) || b.is_a?(Mixin)
-
             res = node_rank(a) <=> node_rank(b)
             res = node_name(a) <=> node_name(b) if res == 0
+            res = (original_order[a] || 0) <=> (original_order[b] || 0) if res == 0
```

Comment:
I am not quite clear how we fallback to this case for mixins? Is it that their node names are always the same? What prevents mixin nodes from being sorted alphabetically?

Reply:
`node_name` always returns `nil` for nodes that shouldn't be sorted alphabetically: https://github.com/Shopify/tapioca/blob/master/lib/tapioca/rbi/rewriters/sort_nodes.rb#L69-L70